woocommerce/woocommerce-android
# Add option in beta settings to switch on/off products

Fixes #1601. This PR adds a new option to the beta section in the Settings page for users to opt in/out of the new products tab.

### Changes:
- Added new products switch to the beta features screen.
- Added new flag in `AppPrefs` to display/hide products tab.
- Removed Feature flags: `PRODUCT_LIST` and `PRODUCT_VARIANTS` since they are both  clubbed into the same release.
- Added a new event `SETTINGS_BETA_FEATURES_PRODUCTS_TOGGLED` which has been added to the events spreadsheet and has been validated/registered.

### Screenshots:
(LEFT: products tab is switched on . RIGHT: products tab is switched off)
<img width="300" src="https://user-images.githubusercontent.com/22608780/69398514-92b20700-0d10-11ea-8799-cd2a75f02679.gif"> . <img width="300" src="https://user-images.githubusercontent.com/22608780/69398516-93e33400-0d10-11ea-88f4-1034b17546d9.gif">

### Testing:
- Verify that products tab is only displayed if the product feature switch in Settings page is switched on.
- Verify that product list is displayed, product search works as expected.
- Verify that clicking on a product redirects to product detail page.
- Notice that variants section is displayed in product detail page, for a variable product.
- Verify that clicking on the variants section redirects to the product variants screen.

Update release notes:
